pkg/capabilities/triggers/mercury_trigger_test.go

```diff
@@ -0,0 +1,201 @@
+package triggers
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/mercury"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+func TestMercuryTrigger(t *testing.T) {
+	ts := NewMercuryTriggerService()
+	ctx := tests.Context(t)
+	require.NotNil(t, ts)
+
+	m := map[string]interface{}{
+		"feedIds":   []int64{1},
+		"triggerId": "test-id-1",
+	}
+
+	wrapped, err := values.NewMap(m)
+	require.NoError(t, err)
+
+	cr := capabilities.CapabilityRequest{
+		Metadata: capabilities.RequestMetadata{
+			WorkflowID: "workflow-id-1",
+		},
+		Inputs: wrapped,
+	}
+	callback := make(chan capabilities.CapabilityResponse, 10)
+	require.NoError(t, ts.RegisterTrigger(ctx, callback, cr))
+
+	// Send events to trigger and check for them in the callback
+	fr := []mercury.FeedReport{
+		{
+			FeedID:               1,
+			FullReport:           []byte("0x1234"),
+			BenchmarkPrice:       2,
+			ObservationTimestamp: 3,
+		},
+	}
+	err = ts.ProcessReport(fr)
+	assert.NoError(t, err)
+	assert.Len(t, callback, 1)
+	select {
+	case msg := <-callback:
+		unwrapped, _ := mercury.Codec{}.UnwrapMercuryTriggerEvent(msg.Value)
+		assert.Equal(t, "mercury", unwrapped.TriggerType)
+		assert.Equal(t, GenerateTriggerEventID(fr), unwrapped.ID)
+		assert.Len(t, unwrapped.Payload, 1)
+		assert.Equal(t, fr[0], unwrapped.Payload[0])
+	default:
+		// The channel is empty; proceed without blocking
```

Comment:
Should we have this at all? We always expect something to be in that channel in this test.